lib/is-semantic-message.js

```diff
@@ -1,22 +1,30 @@
-const commitTypes = Object.keys(require('conventional-commit-types').types)
-const { validate } = require('parse-commit-message')
+const commitTypes = Object.keys(require("conventional-commit-types").types);
+const isScopeValid = require("./is-scope-valid");
+const { validate } = require("parse-commit-message");
 
-module.exports = function isSemanticMessage (message, validScopes, validTypes, allowMergeCommits, allowRevertCommits) {
-  const isMergeCommit = message && message.startsWith('Merge')
-  if (allowMergeCommits && isMergeCommit) return true
+module.exports = function isSemanticMessage(
+  message,
+  validScopes,
+  validTypes,
+  allowMergeCommits,
+  allowRevertCommits
+) {
+  const isMergeCommit = message && message.startsWith("Merge");
+  if (allowMergeCommits && isMergeCommit) return true;
 
-  const isRevertCommit = message && message.startsWith('Revert')
-  if (allowRevertCommits && isRevertCommit) return true
+  const isRevertCommit = message && message.startsWith("Revert");
+  if (allowRevertCommits && isRevertCommit) return true;
 
-  const { error, value: commits } = validate(message, true)
+  const { error, value: commits } = validate(message, true);
 
   if (error) {
-    if (process.env.NODE_ENV !== 'test') console.error(error)
-    return false
+    if (process.env.NODE_ENV !== "test") console.error(error);
+    return false;
   }
 
-  const [result] = commits
-  const { scope: scopes, type } = result.header
-  const isScopeValid = !validScopes || !scopes || scopes.split(',').map(scope => scope.trim()).every(scope => validScopes.includes(scope))
-  return (validTypes || commitTypes).includes(type) && isScopeValid
-}
+  const { type } = result.header;
```

Comment:
`result` is undefined since the line `const [result] = commits;` has been removed.
See: https://github.com/zeke/semantic-pull-requests/pull/124/files#r565929688